### ISSUE TITLE
Seems like there is a bug

### ISSUE THREAD

**Issue by TheGreatAbyss** [crisboarna/terraform-aws-api-gateway-lambda-dynamodb#1]
Hi, 

I'm not an expert in TF so it could be user error, but I get the below errors.

```
➜  **:(master) ✗ terraform plan

Error: Missing required argument

The argument "region" is required, but was not set.


Error: Unsupported block type

  on .terraform/modules/api-gateway-lambda-dynamodb/crisboarna-terraform-aws-api-gateway-lambda-dynamodb-799f299/modules/global/iam/main.tf line 4, in data "template_file" "lambda_dynamodb_policy":
   4:   vars {

Blocks of type "vars" are not expected here. Did you mean to define argument
"vars"? If so, use the equals sign to assign it a value.
```

**Comment by TheGreatAbyss** [comment]
Here's my usage:

```
module "api-gateway-lambda-dynamodb" {
  source  = "crisboarna/api-gateway-lambda-dynamodb/aws"
  version = "1.13.0"
  
  # insert the 9 required variables here
  region = "us-east-1"
  project = var.project
  lambda_runtime = var.lambda_runtime
  lambda_memory_size = var.lambda_memory_size
  lambda_function_name = "My Project"
  lambda_handler = "my_project/lambda_runner.py"

   #DynamoDB
  dynamodb_table_properties = [
    {
      name = "Test Table 1",
      read_capacity = 2,
      write_capacity = 3,
      hash_key = "KEY"
      range_key = ""
      stream_enabled = "true"
      stream_view_type = "NEW_IMAGE"
    }
  ]

  dynamodb_table_attributes = [[
    {
      name = "KEY"
      type = "S"
    }],[
    {
      name = "PRIMARY_KEY"
      type = "N"
    }, {
      name = "SECONDARY_KEY"
      type = "S"
    }
   ]]

    #Lambda Environment variables
  environment_variables = {
    TEST_ENV = "Bernie"
  }

}
```

Thanks!

**Comment by archenroot** [comment]
@TheGreatAbyss - the issue is related to Terraform itself, I hit this as well. You need to create in every module/submodule variable region and best assign default value, in the main.tf just add in first line:
```
provider "aws" {
  region = "${var.region}"
}
```

That is one of temporary workarounds... https://github.com/hashicorp/terraform/issues/21330

**Comment by sowmiya-ragu** [comment]
**_Error:
Blocks of type "environment_variables" are not expected here. Did you mean to
define argument "environment_variables"? If so, use the equals sign to assign_**

**Solution:**
it a value.Below works like a charm - Terraform v0.12.21
 environment_variables **=** {
			hello = "heyy"
                            }
The "=" after enviornmental_varaibles matters.

**Comment by crisboarna** [comment]
@TheGreatAbyss error is caused by him using Terraform 0.12+ whereas this module up to v1.14.0 is compatibile with up to Terraform 0.11.x. Please use recently published module version v1.15.1 for Terraform 0.12 usage.
